src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -2629,6 +2678,111 @@ def test_structures(fortran_reader, fortran_writer):
         "  PROCEDURE, NOPASS :: test_code\n"
         "END TYPE test_type\n" in result)
 
+    # type that contains procedures
+    test_code = (
+        "module test_mod\n"
+        "  use kernel_mod, only : parent_type\n"
+        "  type, extends(parent_type) :: test_type\n"
+        "    integer :: i = 1\n"
+        "    contains\n"
```

Comment:
Please could you extend this test so that the `type` contains a `private` statement. The visibility of the `test_code` procedure should then become `private`.

Reply:
Done, it works :)

Reply:
Not quite and unfortunately I think it's broken (see below) ! :-)